plotsearch/serializers/plot_search.py

```diff
@@ -352,12 +352,12 @@ def get_plan_unit_or_custom_detailed_plan(self, validated_data):
         plan_unit = None
         try:
             plan_unit = PlanUnit.objects.get(id=validated_data.pop("plan_unit_id"))
-        except KeyError:
+        except (KeyError, PlanUnit.DoesNotExist):
             try:
                 custom_detailed_plan = CustomDetailedPlan.objects.get(
                     id=validated_data.pop("custom_detailed_plan_id")
                 )
-            except KeyError:
+            except (KeyError, CustomDetailedPlan.DoesNotExist):
```

Comment:
Should you refactor this whole function and not just add these DoesNotExist exceptions?

Reply:
Let me look into it in a bit more detail, since now there's 3 tracks that are being checked: The validated_data, the actual objects existance, and then these interim variables that are defined None at first. So there could be a chance to make it more clear

Reply:
@antti-honkanen How does it looke like now? I tried to make it simpler.
The reason why the `KeyError` does not trigger is probably because the field has been allowed to be null, so the original code was not actually working